solana/solana-ibc/programs/solana-ibc/src/transfer/mod.rs

```diff
@@ -142,12 +144,115 @@ impl ibc::Module for IbcStorage<'_, '_> {
             .into_bytes(),
             ..packet.clone()
         };
-        let (extras, ack) = ibc::apps::transfer::module::on_recv_packet_execute(
-            self,
-            &maybe_ft_packet,
-        );
-        let ack_status = str::from_utf8(ack.as_bytes())
+        let (extras, mut ack) =
+            ibc::apps::transfer::module::on_recv_packet_execute(
+                self,
+                &maybe_ft_packet,
+            );
+        let cloned_ack = ack.clone();
+        let ack_status = str::from_utf8(cloned_ack.as_bytes())
             .expect("Invalid acknowledgement string");
+        let status = serde_json::from_slice::<ibc::AcknowledgementStatus>(
+            ack.as_bytes(),
+        );
```

Comment:
```suggestion
        let status = serde_json::from_str::<ibc::AcknowledgementStatus>(
            ack_status
        );
```
Either this or get rid of `ack_status` altogether.